promql/promql-test-queries.yml

```diff
@@ -52,6 +52,7 @@ test_cases:
   - query: 'demo_memory_usage_bytes{instance!~".*:10000"}'
   - query: 'demo_memory_usage_bytes{type="free", instance!="demo.promlabs.com:10000"}'
   - query: '{type="free", instance!="demo.promlabs.com:10000"}'
+  - query: 'promhttp_metric_handler_requests_in_flight and ignoring(code) promhttp_metric_handler_requests_total'
```

Comment:
Thanks! A couple of points:

* This should go somewhere in the binary operator test queries section: https://github.com/prometheus/compliance/blob/d128251d4ca612e9782fffd9a8e3c9f9d6f59581/promql/promql-test-queries.yml#L99-L134
* That section is currently woefully insufficient. It hasn't been testing `ignoring` at all (only `on`), so maybe adding similar queries with `ignoring` there as the currently existing `on` ones could already uncover this issue as well.
* Currently the testing setup only scrapes data from the demo service (https://github.com/juliusv/prometheus_demo_service), and new queries should stick to that as well. You can play with all the `demo_` metrics here: https://demo.promlabs.com/graph.

Reply:
Ah right, it does get emitted as part of the standard client library metrics from the demo job, but I'd prefer to stick to the metrics that are explicitly created by the demo service itself if we can. So far all other examples only query those `demo_` metrics.